app/utils/detection.js

```diff
@@ -0,0 +1,131 @@
+import UAParser from 'ua-parser-js';
+import semverCompare from 'semver-compare';
+import identity from 'lodash/identity';
+
+const DEFAULT_SUPPORTED_BROWSERS_CONFIG = {
+  desktop: [{
+    browser: 'firefox', minversion: 41,
+  }, {
+    browser: 'ie', versions: [11, 'edge'],
+  }, {
+    browser: 'chrome', minversion: 45,
+  }, {
+    browser: 'edge',
+  }, {
+    os: 'mac os', minos: '10.10.0', browser: 'safari', minversion: 8,
+  }],
+  tablet: [{
+    os: 'ios', minos: '9', browser: 'mobile safari',
+  }, {
+    os: 'android', minos: '5.0', browser: 'chrome',
+  }, {
+    browser: 'ie', versions: [11, 'edge'],
+  }, {
+    browser: 'edge',
+  }],
+  mobile: [{
+    os: 'ios', minos: '9', browser: 'mobile safari',
+  }, {
+    os: 'ios', minos: '5.0', browser: 'chrome',
+  }, {
+    os: 'android', minos: '5.0', browser: 'chrome', minversion: 50,
+  }],
+};
+
+export default class Detection {
```

Comment:
`Detection` is too broad. Can you name it differently? e.g. `UnsupportedBrowserDetection` ?